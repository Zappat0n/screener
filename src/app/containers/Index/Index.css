@import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@400;700&display=swap');

button {
  background: white;
  border: none;
}

.index button {
  margin: 1rem;
  border: 1px solid lightgray;
  outline: none;
  text-shadow: 0 0 1px #000;
  box-shadow: 0 0 10px lightgray;
  font: normal normal 600 1rem/2rem 'Open Sans', sans-serif;
  border-radius: 1rem;
  padding: 0.5rem;
}

button:hover {
  cursor: pointer;
  background: lightgray;
}

.backgroundRed {
  color: red;
}

.backgroundBlue {
  color: blue;
}
