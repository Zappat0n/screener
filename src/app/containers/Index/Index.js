import { useSelector } from "react-redux";

const Index = (props) => {
  const { ticker } = props;
  const selectName = useSelector(state => {
    return state.indexes[ticker].name;
  });
  const selectPrice = useSelector(state => state.indexes[ticker].price);
  const selectChange = useSelector(state => state.indexes[ticker].changesPercentage);

  return (
    <h4>
      {selectName} : {selectPrice} {selectChange}%
    </h4>
  );
}

export default Index;
