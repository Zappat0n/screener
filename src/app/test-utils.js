import React from 'react';
import { createStore, combineReducers } from 'redux';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import indexReducer from './reducers/indexSlice';
import stockReducer from './reducers/stockSlice';
import profileReducer from './reducers/profileSlice';
import filterReducer from './reducers/filterSlice';

const rootReducer = combineReducers({
  indexes: indexReducer, stocks: stockReducer, profile: profileReducer, filter: filterReducer,
});

const initialState = {
  stocks: {
    AAPL: {
      price: 100,
      companyName: 'Apple Inc',
      changes: 1,
      sector: 'Technology',
    },
  },
  profile: {
    data: {
      image: '',
      companyName: 'Microsoft',
      price: 100,
      changes: 1,
      mktCap: 1000000,
      range: '0-100',
      country: 'US',
    },
  },
};

const renderComponent = (component) => render(
  <Provider store={createStore(rootReducer, initialState)}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
);

export * from '@testing-library/react';
export { renderComponent };
