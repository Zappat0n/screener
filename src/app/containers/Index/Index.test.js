import React from 'react';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import Index from './Index';
import { renderComponent } from '../../test-utils';

renderComponent(
  <Index ticker="GSPC" />,
);

test('Renders Index correctly', () => {
  expect(screen.getByText('S&P 500')).toBeInTheDocument();
});
