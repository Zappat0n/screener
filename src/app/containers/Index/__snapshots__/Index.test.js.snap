// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Match SnapShot 1`] = `
<div
  className="index"
>
  <button
    type="button"
  >
    S&P 500
    <div
      className="colorBlue"
    >
      4000
       
      2
      %
    </div>
  </button>
</div>
`;
